assets/aml-benchmark/components/prompt_crafter/spec.yaml

```diff
@@ -0,0 +1,107 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+type: command
+
+name: prompt_crafter_
+display_name: BenchmarkingPromptCrafter
+description: Create prompts from a given dataset. From a given jinja prompt 
+  template, it will generate prompts. It can also create few-shot prompts 
+  given a few-shot dataset and the number of shots.
+version: 0.0.1.0
+is_deterministic: true
+
+inputs:
+  prompt_type:
+    type: string
+    optional: false
+    enum:
+      - chat
+      - completions
+    description: |
+      Determine the prompt format. This component supports chat and completion apis.
+      Completion: 
+      {"prompt": "Few shot prompts will go here"}
+      Chat: 
+      Chat models have 3 roles: System, User and Assistant. 
+      Example:
+      {"prompt": [{"role": "system", "content": "You are a helpful assistant"},
+                    {"role": "user", "content": "Example chat input"},
+                    {"role": "assistant", "content": "Example chat output"}]}
+    default: completions
+  test_data:
+    type: uri_file
+    optional: false
+    description: |
+      The uri file (jsonl) used to generate prompts.
+  prompt_pattern:
+    type: string
+    optional: false
+    description: |
+      The pattern to be used to generate the prompts. It should be a valid jinja template. 
+
+      Example:
+      Input Data : 
+      {"question":"Example Question?",
+      "choices":{"option":["Answer1","Answer2","Answer3","Answer4"]},
+      "answerKey":"D"}
+
+      The prompt pattern for the above input data can be:
+      "Question: {{question}}\n
+      Choices are: 
+      (1) {{choices.option[0]}}\n
+      (2) {{choices.option[1]}}\n
+      (3) {{choices.option[2]}}\n
+      (4) {{choices.option[3]}}\n"
+  n_shots:
+    type: integer
+    optional: false
+    default: 0
+    description: |
+      The number of shots to use in the few-shot prompts. Default is 0, which means no few-shot
+      examples will be generated. n_shots must be smaller than the size of few_shot dataset.
+  few_shot_data:
+    type: uri_file
+    optional: true
+    description: |
+      The uri file(jsonl) to be used to generate the n-shot prompts.
+  output_pattern:
+    type: string
+    optional: false
+    description: |
+      The jinja template representing for the ground truth. For n-shot prompts>0, this will be added
```

Comment:
* Can "The jinja template representing for the ground truth." simply be "The jinja template for the ground truth."?
* Can `prompts>0` be `prompts > 0` (spaces around it)?

Reply:
Updated. Did not mention "ground_truth" word as we might add the ground_truth/label_key columns to be used as is for cases where there are >1 ground_truths, etc.

Reply:
Updated. Avoided the word ground_truth but mentioned it as expected output so that in case we add label_key column in the future, there are no confusion.